app/models/product.rb

```diff
@@ -1,20 +1,47 @@
 class Product < ApplicationRecord
+  PERMALINK_REGEX = /[[:alnum:]]+/
+  validates_with ImageUrlValidator, attributes: [:image_url], if: ->{ image_url.present? }
```

Comment:
Lets use `url: true` format and each validator for this validation as required.

Reply:
Done